src/MetaOptions.js

```diff
@@ -8,11 +8,16 @@ const BODY_KEY_CASE_OPTIONS = {
 
 export default class MetaOptions {
   constructor(options) {
-    if (!options.bodyKeyCase || !this.isValidBodyKeyCase(options.bodyKeyCase)) {
+    if (!this.isValidBodyKeyCase(options.bodyKeyCase)) {
       throw new Error('This body key formatting option is not allowed.')
     }
-
     this.bodyKeyCase = options.bodyKeyCase
+
+    if (typeof options.logger === 'function') {
+      this.logger = options.logger
+    } else {
+      this.logger = () => {} // no-op logger
+    }
```

Comment:
Couldn't we add `() => {}` as the logger in the `defaultMetaOptions` in the `index.js` and in here just validate if any valid logger was passed at all?

Since we are using `this.metaOptions = new MetaOptions({...defaultMetaOptions, ...metaOptions})`, the default logger should be overridden if one is passed in, shouldn't it?

Reply:
Sounds good